### PR TITLE
Add edxapp configuration options for user retirement

### PR DESCRIPTION
**Pull request by bmedx**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [x] A DevOps team member has approved the PR.
  - [x] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [x] Update the appropriate internal repo (be sure to update for all our environments)
    - [x] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [x] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+- Role: edxapp
+  - Added `EDXAPP_RETIRED_USERNAME_FMT`, `EDXAPP_RETIRED_EMAIL_FMT`, `EDXAPP_RETIRED_USER_SALTS`, and 
+  `EDXAPP_RETIREMENT_SERVICE_WORKER_USERNAME` to generic_env_config to allow user retirement to be configurable.
```
Comment (jibsheet) [inline CHANGELOG.md:3]: I misread this earlier and asked about env vs auth.  I meant to ask:
This should be available to lms and cms, right?

**Review by jibsheet — APPROVED** [review]
If it's supposed to go in both {lms,cms}.env.json then ship it

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -719,6 +719,13 @@ EDXAPP_PROCTORING_BACKEND_PROVIDER:
   class: 'edx_proctoring.backends.null.NullBackendProvider'
   options: {}
 
+EDXAPP_RETIRED_USERNAME_FMT: "retired__user_{}"
+EDXAPP_RETIRED_EMAIL_FMT: "retired__user_{}@retired.invalid"
+EDXAPP_RETIRED_USER_SALTS:
+  - "OVERRIDE ME WITH A RANDOM VALUE"
+  - "ROTATE SALTS BY APPENDING NEW VALUES"
```
Comment (jibsheet) [inline playbooks/roles/edxapp/defaults/main.yml:726]: Do we have any suggestions on making these for yourself?  I realize cryptography is less important here, but I'm lazy so I used pwgen which gets ms a-zA-Z0-9 in 20 characters.  Does it need to be base64/base62?  I assume this is somewhere in the tickets I haven't read yet.

Comment (bmedx) [reply playbooks/roles/edxapp/defaults/main.yml:726]: It's not in the tickets, a-zA-Z-9 x 20 should be fine. Base62/4 is also good. Python just needs to get it as a string in settings.

File: CHANGELOG.md (hunk 1 above)
Comment (bmedx) [reply CHANGELOG.md:3]: We'd like to be able to reach use these in both, I attempted to put it in the right place to do that here. Let me know if I messed up.

Comment (jibsheet) [reply CHANGELOG.md:3]: 👍

File: playbooks/roles/edxapp/defaults/main.yml (hunk 2 above)
Comment (jibsheet) [reply playbooks/roles/edxapp/defaults/main.yml:726]: 👍

**Review by doctoryes — APPROVED** [review]

**Pull request merged**